pkg/diagnostics/log/log.go

```diff
@@ -48,27 +64,64 @@ func NewLogger(setLevel int, setFormat string, out io.Writer) (*Logger, error) {
 	var err error = nil
 	level := DebugLevel
 	switch setLevel {
-	case 0:
+	case ErrorLevel.Level:
 		level = ErrorLevel
-	case 1:
+	case WarnLevel.Level:
 		level = WarnLevel
-	case 2:
+	case NoticeLevel.Level:
 		level = NoticeLevel
-	case 3:
+	case InfoLevel.Level:
 		level = InfoLevel
-	case 4:
+	case DebugLevel.Level:
 		// Debug, also default for invalid numbers below
 	default:
 		err = errors.New("Invalid diagnostic level; must be 0-4")
 	}
+
 	return &Logger{
-		logger: logger,
-		level:  level,
+		loggerType: logger,
+		level:      level,
 	}, err
 }
 
-// a map message type to throw type safety and method signatures out the window:
-type Msg map[string]interface{}
+type Message struct {
```

Comment:
I mostly like what you did with logging, formalizing the parts of the message that didn't vary much.